pennylane/gradients/pulse_generator_gradient.py

```diff
@@ -421,8 +420,262 @@ def processing_fn(results):
+
+        The coefficients :math:`\omega_{k}^{(\ell)}` of the generators can be computed
+        by decomposing the anti-Hermitian matrix :math:`\Omega_{k}` into the Pauli
+        basis and only keeping the non-vanishing terms. This is possible via a tensor
+        contraction with the full Pauli basis (or alternative, more efficient methods):
+
+        .. math::
+
+            \omega_{k}^{(\ell)} = \frac{1}{2^N}\mathrm{Tr}\left[P_\ell \Omega_{k}\right]
+
+        where :math:`N` is the number of qubits.
+
+        Thus far, we discussed the derivative of the time evolution, or pulse.
+        Now, consider an objective function that is based on measuring an expectation
+        value after executing a pulse program:
+
+        .. math::
+            C(\boldsymbol{\theta})=
+            \langle\psi_0|U(\boldsymbol{\theta})^\dagger B
+            U(\boldsymbol{\theta}) |\psi_0\rangle
+
+        Using the derivative of :math:`U` and the decomposition of the effective
+        generator :math:`\Omega_k` above, we calculate the partial derivative of
+        :math:`C`:
+
+        .. math::
+
+            \frac{\partial C}{\partial \theta_{k}} (\boldsymbol{\theta})&=
+            \langle\psi_0|\left[U^\dagger B U, \Omega_{k}\right]|\psi_0\rangle\\
+            &=\sum_{\ell=1}^L \omega_{k}^{(\ell)}
+            \langle\psi_0|\left[U^\dagger B U, P_\ell \right]|\psi_0\rangle\\
+            &=\sum_{\ell=1}^L \tilde\omega_{k}^{(\ell)}
+            \langle\psi_0|\left[U^\dagger B U, -\frac{i}{2}P_\ell \right]|\psi_0\rangle\\
+            &=\sum_{\ell=1}^L \tilde\omega_{k}^{(\ell)}
+            \frac{\mathrm{d}}{\mathrm{d}x}
+            \langle\psi_0|\exp\left(i\frac{x}{2}P_\ell \right)U^\dagger B
+            U\exp\left(-i\frac{x}{2}P_\ell \right)|\psi_0\rangle\large|_{x=0}\\
+            &=\sum_{\ell=1}^L \tilde\omega_{k}^{(\ell)}
+            \frac{\mathrm{d}}{\mathrm{d}x} C_\ell(x)\large|_{x=0}
+
+        where we skipped the argument :math:`\boldsymbol{\theta}` of :math:`U` for readability
+        and introduced the modified coefficients
+        :math:`\tilde\omega_{k}^{(\ell)}=2i\omega_{k}^{(\ell)}`.
+        In the second to last step, we rewrote the commutator of :math:`U^\dagger BU` and
+        :math:`\frac{i}{2}P_\ell` as the derivative (at zero) of a modified cost function
+        :math:`C_\ell(x)` that executes a Pauli rotation about :math:`-i\frac{x}{2}P_\ell`
+        before the parametrized time evolution. Here, the variable :math:`x` is just a
+        convenient way to write the modified cost function. Note that its derivative with
+        respect to :math:`x` can be computed with the standard two-term parameter-shift
+        rule for Pauli rotation gates.
+
+        *Caching*
+
+        Considering the derivation above, we notice that the same modified cost function
+        :math:`C_\ell(x)` may appear in the derivatives of distinct parameters
+        :math:`\theta_k` and :math:`\theta_m`. In order to not evaluate the same
```

Comment:
```suggestion
        :math:`\theta_k` and :math:`\theta_m`, because they both enter into the same
        pulse gate. In order to not evaluate the same
```